Nullable enums cause NPE

Hello,
I was trying to generate the client and model code for the German Brightsky API: https://brightsky.dev/docs/

Their OpenAPI Spec works flawlessly in the Swagger Editor, but when trying to generate the code with fabrikt, it simply fails due to 

```
Caused by: java.lang.NullPointerException: Cannot invoke "Object.toString()" because "it" is null
	at com.cjbooms.fabrikt.util.KaizenParserExtensions.getEnumValues(KaizenParserExtensions.kt:94)
	at com.cjbooms.fabrikt.model.KotlinTypeInfo$Companion.from(KotlinTypeInfo.kt:96)
	at com.cjbooms.fabrikt.model.PropertyInfo$Field.<init>(PropertyInfo.kt:170)
	at com.cjbooms.fabrikt.model.PropertyInfo$Companion.getInLinedProperties(PropertyInfo.kt:127)
	at com.cjbooms.fabrikt.model.PropertyInfo$Companion.topLevelProperties(PropertyInfo.kt:57)
	at com.cjbooms.fabrikt.generators.model.ModelGenerator.createModels(ModelGenerator.kt:179)
	at com.cjbooms.fabrikt.generators.model.ModelGenerator.generate(ModelGenerator.kt:162)
	at com.cjbooms.fabrikt.cli.CodeGenerator.models(CodeGenerator.kt:65)
	at com.cjbooms.fabrikt.cli.CodeGenerator.generateClient(CodeGenerator.kt:55)
	at com.cjbooms.fabrikt.cli.CodeGenerator.generateCode(CodeGenerator.kt:36)
	at com.cjbooms.fabrikt.cli.CodeGenerator.generate(CodeGenerator.kt:32)
	at com.cjbooms.fabrikt.cli.CodeGen.generate(CodeGen.kt:84)
	at com.cjbooms.fabrikt.cli.CodeGen.main(CodeGen.kt:22)
	at ch.acanda.gradle.fabrikt.generator.FabriktGeneratorKt.generate(FabriktGenerator.kt:18)
	... 119 more
```

When I removed all `null` values from all the enums in the YAML it worked.

A quick debugging session showed that the issue is there: https://github.com/cjbooms/fabrikt/blob/d538e020cf12f7eaacd09bf1c117363c77cb8afb/src/main/kotlin/com/cjbooms/fabrikt/util/KaizenParserExtensions.kt#L94

So that `toString()` call there needs to use the correct extension function for `Any?`, and not the member method of `Any`. I don't know if that is enough, though.

BUT: Even after removing all nullable enums, the generated client code (for OkHTTP) did not compile still. It tried to provide a `List<Int>` to a function that only accepts `List<String>`. That is the case for all the `source_id` parameters in the API spec.

PS: I was using a Gradle build with the latest version 1.11.0 of the plugin, which in turn uses fabrikt 20.0.0.

--- a/src/main/kotlin/com/cjbooms/fabrikt/util/KaizenParserExtensions.kt
+++ b/src/main/kotlin/com/cjbooms/fabrikt/util/KaizenParserExtensions.kt
@@ -91,7 +91,7 @@ object KaizenParserExtensions {
 
     @Suppress("UNCHECKED_CAST")
     fun Schema.getEnumValues(): List<String> = when {
-        this.hasEnums() -> this.enums.map { it.toString() }.filterNot { it.isBlank() }
+        this.hasEnums() -> this.enums.filterNotNull().map { it.toString() }.filterNot { it.isBlank() }
         !MutableSettings.modelOptions().contains(ModelCodeGenOptionType.X_EXTENSIBLE_ENUMS) -> emptyList()
         else -> extensions[EXTENSIBLE_ENUM_KEY]?.let { it as List<String?> }?.filterNotNull()
             ?.filterNot { it.isBlank() } ?: emptyList()

--- a/src/test/resources/examples/enumExamples/api.yaml
+++ b/src/test/resources/examples/enumExamples/api.yaml
@@ -33,6 +33,8 @@ components:
           $ref: '#/components/schemas/ExtensibleEnumObject'
         list_enums:
           $ref: '#/components/schemas/ListEnums'
+        nullable_ref:
+          $ref: '#/components/schemas/NullableEnum'
 
     EnumObject:
       type: string
@@ -85,5 +87,7 @@ components:
           bar_prop:
             type: string
 
-
-
+    NullableEnum:
+      type: string
+      nullable: true
+      enum: [ "foo", "bar", null ]

--- a/src/test/resources/examples/enumExamples/models/EnumHolder.kt
+++ b/src/test/resources/examples/enumExamples/models/EnumHolder.kt
@@ -22,4 +22,7 @@ public data class EnumHolder(
   @param:JsonProperty("list_enums")
   @get:JsonProperty("list_enums")
   public val listEnums: List<ContentType>? = null,
+  @param:JsonProperty("nullable_ref")
+  @get:JsonProperty("nullable_ref")
+  public val nullableRef: NullableEnum? = null,
 )

--- a/src/test/resources/examples/enumExamples/models/NullableEnum.kt
+++ b/src/test/resources/examples/enumExamples/models/NullableEnum.kt
@@ -1,0 +1,20 @@
+package examples.enumExamples.models
+
+import com.fasterxml.jackson.`annotation`.JsonValue
+import kotlin.String
+import kotlin.collections.Map
+
+public enum class NullableEnum(
+  @JsonValue
+  public val `value`: String,
+) {
+  FOO("foo"),
+  BAR("bar"),
+  ;
+
+  public companion object {
+    private val mapping: Map<String, NullableEnum> = entries.associateBy(NullableEnum::value)
+
+    public fun fromValue(`value`: String): NullableEnum? = mapping[value]
+  }
+}

